tornado-api/src/main/java/uk/ac/manchester/tornado/api/types/tensors/Tensor.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2024, APT Group, Department of Computer Science,
+ * The University of Manchester.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.manchester.tornado.api.types.tensors;
+
+import uk.ac.manchester.tornado.api.types.arrays.TornadoNativeArray;
+
+import java.lang.foreign.MemorySegment;
+
+public non-sealed class Tensor extends TornadoNativeArray {
+    @Override
+    public int getSize() {
+        return 0;
```

Comment:
The implementation of this class is empty. This class should invoke to the child implementation, so probably you need a Tensor type in the constructor and keep a field with the implementation, or build one your own based on the DType. 
I think the second option is the one to go